src/main/java/org/example/intro/service/impl/BookServiceImpl.java

```diff
@@ -49,15 +47,14 @@ public void delete(Long id) {
     }
 
     @Override
-    public List<BookDto> search(BookSearchParametersDto params, int page) {
-        Pageable pageable = PageRequest.of(page, ITEMS_PER_PAGE);
+    public List<BookDto> search(BookSearchParametersDto params, Pageable pageable) {
```

Comment:
Have you done the HW to implement search endpoint?

Reply:
This HW was related to "findAll" controller method.

But i decided to improve this method as well. I made this controller method in the previous PR. But here after watching new video lesson i thought to improve this method. I removed the path variable, where i put the page number before and updated like with normal fetching of all books - with using Pageable class.

Reply:
It's better to implement search endpoint according to the **Add ability to search books** HW (see Spring Boot / Criteria Query in Spring Boot section)

Reply:
it seems i don't understand what to make here.
could you please elaborate what issues here?

Reply:
<img width="692" alt="Screenshot 2024-07-24 at 13 14 37" src="https://github.com/user-attachments/assets/4551577d-0912-47b5-a1de-0e7119191dfb">
Please, check this module and implementation of search endpoint. I didn't find any of specifications in your code

Reply:
++, Thank you! Updated. I merged by mistake the PR where i wanted to make the "search" method. and made changes here - i didn't know how to restore that PR.